### ISSUE TITLE
Using Jax XLA code

### ISSUE THREAD

**Issue by GJBoth** [elixir-nx/nx#1322]
I'm very excited to see XLA support in Elixir, but NX doesn't support everything I do yet. Would it be possible to write the "numerical" part of my code in Jax and [compile it AOT](https://jax.readthedocs.io/en/latest/aot.html#inspecting-staged-out-computations), and then point the Elixir XLA client at the XLA cache generated by Jax?

**Comment by josevalim** [comment]
We have no plans to explore something along the lines of running JAX AOTs. However, if there is a particular XLA feature that you are missing, then let us know, and we will investigate adding parity!

**Comment by GJBoth** [comment]
Thanks for the quick response! Okay, that's a shame. One of the things I miss for example (maybe I misread the documentation?) is 2D FFTs? And what about adding custom gradients (similar to `custom_jvp` and and `custom_vjp`)?

**Comment by polvalente** [comment]
> Thanks for the quick response! Okay, that's a shame. One of the things I miss for example (maybe I misread the documentation?) is 2D FFTs? And what about adding custom gradients (similar to `custom_jvp` and and `custom_vjp`)?

2D FFTs are indeed missing

Custom grads are already possible inside defn: https://hexdocs.pm/nx/Nx.Defn.Kernel.html#custom_grad/3

**Comment by josevalim** [comment]
New issue here: https://github.com/elixir-nx/nx/issues/1323 :)
